### PR TITLE
Search bar in top bar

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Moved search bar from app bar into top bar

## Screenshots (if appropriate):
host docker internal_8300_index html(iPad Pro)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by LukasHirt** [comment]
For now, it is built on top of the [sidebar PR](https://github.com/owncloud/phoenix/pull/3442#issuecomment-627920838). After it'll get merged, I'll drop that commit in here.

**Comment by LukasHirt** [comment]
There is one issue with search on mobile. Since we hide it in the drop, this results in the search being still visible after the search has been finished. However, if we use the `closeOnClick` then it works if we execute the search by clicking on the search button but it also hides it if we click only into the input. And of course, if the search is executed by via keyboard then the result is the same as in the beginning.

It's times like these I really wish for having drop with a simple `v-if` instead of the UIkit one...

**Comment by LukasHirt** [comment]
Fixed the selector in tests for loading spinner in the search bar. Tests are now passing locally. Also, I've hidden the top bar in public links on wider screens because they had no content there

**Comment by LukasHirt** [comment]
Ok, one more change - hidden top bar and sidebar in public pages. Top bar because there is no element which could be used (opened a ticket to make search work in public links #3521) and sidebar because there are no nav items (opened a ticket for branding in such case #3520)

**Comment by LukasHirt** [comment]
Tests are green. Pls review @PVince81 @kulmann

File: src/components/SearchBar.vue [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      :button-hidden="true"
+      @search="search"
+      @clear="resetSearch"
+    />
+    <oc-button
+      id="files-open-search-btn"
+      key="mobile-search-button"
+      class="uk-hidden@m"
+      icon="search"
+      aria-label="search"
```
Comment (kulmann) [inline src/components/SearchBar.vue:17]: aria-label should have translated content

File: src/components/SearchBar.vue [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      this.clearSearchBarValue()
+    }
+  },
+
+  methods: {
+    ...mapActions('Files', ['searchForFile', 'resetSearch']),
+
+    search(searchTerm) {
+      if (!searchTerm) {
+        return
```
Comment (kulmann) [inline src/components/SearchBar.vue:65]: Since typing `enter` emits `search`, it would make sense to call `resetSearch` here just above the return.

**Comment by LukasHirt** [comment]
@kulmann Label translated to "Open search bar" and search reset when hitting enter added.

**Pull request merged**